app/src/check-permissions.ts

```diff
@@ -0,0 +1,32 @@
+import api, { route } from '@forge/api';
+
+// Forge does not export their Request type, so we have to resort to 'any' for now.
+export const adminPermissionCheck = async (req: any): Promise<void> => {
+	const { accountId } = req.context;
+
+	const permissionRequestBody = `{
+		"globalPermissions": [
+			"ADMINISTER"
+		],
+		"accountId": "${accountId}"
+	}`;
+
+	const permissions = await api.asUser().requestJira(route`/rest/api/3/permissions/check`, {
+		method: 'POST',
+		headers: {
+			Accept: 'application/json',
+			'Content-Type': 'application/json'
+		},
+		body: permissionRequestBody
+	});
+
+	if (permissions.status === 403) {
+		throw new Error('Only Jira administrators can perform this operation.');
+	}
+
+	const permissionDetails = await permissions.json();
+
+	if (!permissionDetails?.globalPermissions?.includes('ADMINISTER')) {
+		throw new Error('Only Jira administrators can perform this operation.');
```

Comment:
This error message is the exact same wording as the error message that Jira returns on a 403, if I remember correctly from our previous discussion. This means that if anyone sees this message in the logs, they won't know whether it comes from Jira or from our app :). I would suggest to rephrase this to make clear that the error is coming from the app: "Only Jira administrators can access the Jenkins for Jira admin page".

Reply:
Yep, I had it like that intentionally initially so we didn't need to worry about getting copy approval. Happy to update it :)